Add multinode role-adding instructions (#562)
timescale/docs.timescale.com-content: getting-started/setup-multi-node-basic.md

- One or more data nodes to store and organize distributed data.     

All nodes begin as single-node TimescaleDBs, i.e., hosts with a running 
PostgreSQL instance and a loaded TimescaleDB extension. This is assumed for 
"access node" and "data node" in the instructions. More detail on 
the architecture can be found in the [Architecture][architecture] section.

> The multi-node can be created as self-managed or hosted on Timescale Cloud or Forge.  In
> order to set up a self-managed cluster, please follow [these instructions][advanced setup]

Reviewer: ```suggestion
TimescaleDB multi-node can be created as part of a self-managed deployment
or (coming soon) as a managed cloud deployment.  In order to set up a
self-managed cluster, including how to configure the nodes for secure
communication and creating users/roles across servers, please follow
[these instructions][advanced setup].
```

Author: changed

> In the case of Timescale Cloud and Forge the created services will already contain

Author: Fixed

> In the case of Timescale Cloud and Forge the created services contain
> PostgreSQL with TimescaleDB loaded and the created user `tsdbadmin` as superuser.
> In this case, all you need to do is decide which service should be the access node,
> and follow the instructions in the next section.

Reviewer: ```suggestion
In the case of Timescale Cloud and Forge, the created services already contain
PostgreSQL with TimescaleDB loaded and the created user `tsdbadmin` as superuser.
In this case, all you will need to do is decide which service should be the access
node, and follow the instructions in the next section.  More information will be
forthcoming as TimescaleDB multi-node is made available on these cloud platforms.
```

Author: changed

The system should be up and running at this point.

## (Optional) Add user roles to the distributed database

After adding your data nodes, it is possible to create and use distributed hypertables,

> but only if running as a superuser common to all nodes. For Timescale Cloud or Forge,
> this means using the created `tsdbadmin` user.

Reviewer: Can hold off until feature launched/available.
```suggestion
but only if running as a superuser common to all nodes (e.g., by default,
the `postgres` user).
```

Author: fixed

If using a self-managed multi-node cluster, you may wish to create new users with
multi-node access, or add access to existing roles.  Please refer to the following
sections depending on the authentication mechanism you are using:
- Adding roles using [trust authentication][trust_role_setup]
- Adding roles using [password authentication][password_role_setup]

> - Adding roles using [certificate authentication][certificate_role_setup]

Reviewer 2: This section feels odd for Forge and Cloud users, who will stay in the document. I understand that it is for self-managed users who will jump into this file for the add data nodes instructions.

Author: Agreed, but it is important for the self-managed users, and hopefully we'll have a way to add users in Forge eventually.  At that point, this section can be expanded to explain how to add setup those roles for multinode.

[install]: /getting-started/installation
[setup]: /getting-started/setup
[advanced setup]: /getting-started/setup-multi-node-basic/setup-multi-node-auth
[trust_role_setup]: /getting-started/setup-multi-node-basic/setup-multi-node-auth#multi-node-auth-trust-roles
[password_role_setup]: /getting-started/setup-multi-node-basic/setup-multi-node-auth#multi-node-auth-password-roles

> [certificate_role_setup]: /getting-started/setup-multi-node-basic/setup-multi-node-auth#multi-node-auth-certificate-roles

Reviewer: Minor point, but should we rename these:

`getting-started/setup-multi-node` and `getting-started/setup-multi-node/auth` ?

Author: It looks like there's already a zombie setup-multi-node page still lingering in this directory that was split into these two new pages (setup-multi-node-basic and setup-multi-node-auth).  @solugebefola did this refactoring about three weeks back, so I'd like to get his feedback here before mucking things up too much.

The system should be up and running at this point.

### 3. (Optional) Add user roles to the distributed database

After the first two steps, it is possible to create and use distributed hypertables,

> but only if running as a superuser common to both nodes. In order to allow additional

Reviewer 3: ```suggestion
but only if running as a superuser common to all nodes. In order to allow additional
```

Author: Fixed

roles to operate on distributed hypertables, they must be properly set up on all the
data nodes.

First, create the role on the access node if not already present:
```sql
CREATE ROLE testrole;
```

Next, allow that role to access the foreign server objects for the data nodes. Run
the following, making sure to include all data node names:
```sql
GRANT USAGE ON FOREIGN SERVER <data node name>, <data node name>, ... TO testrole;
```

Finally add the role to all of the data nodes. Use the [distributed_exec command][distributed_exec]
to do this from the access node:
```sql
CALL distributed_exec($$ CREATE USER testrole WITH LOGIN $$);
```

> It's important that the role have LOGIN permission on the data nodes, even if it

Reviewer 3: ```suggestion
It's important that the role be created with LOGIN permission on the data nodes, even if it
```

Author: Fixed
